projector-plugin/build.gradle.kts

```
@@ -44,3 +45,32 @@ intellij {
 (tasks["runIde"] as JavaExec).apply {
   jvmArgs = jvmArgs.orEmpty() + listOf("-Djdk.attach.allowAttachSelf=true", "-Dswing.bufferPerWindow=false")
 }
+
+abstract class GenerateVersionsFile: DefaultTask() {
+  private val filePath = "src/main/resources/META-INF/versions.txt"
+  @get:Input
+  abstract val agentVersion: Property<String>
+
+  @TaskAction
+  fun generateVersions() {
+    val fullPath = Paths.get(project.buildFile.parent, filePath)
+    val f = File(fullPath.toString())
+    f.printWriter().use {
+      it.println(getContent())
+    }
+  }
+
+  private fun getContent(): String {
+    val versions = listOf("agentVersion=${agentVersion.get()}")
+    return versions.joinToString("\n")
+  }
+}
+
+tasks.register<GenerateVersionsFile>("versions")
+{
+  agentVersion.set(project(":projector-agent").version.toString())
+}
+
+tasks.processResources {
+  dependsOn(tasks["versions"])
+}
```

1. Not sure why `register` is needed.
2. Code style (bracket on the same line).
3. Reuse task object to avoid hardcoding task name in multiple places.

```suggestion
val versions = task<GenerateVersionsFile>("versions") {
  agentVersion.set(project(":projector-agent").version.toString())
}

tasks.processResources {
  dependsOn(versions)
}
```

---

Agreed